url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/step_out.java

```diff
@@ -0,0 +1,220 @@
+package backend;
+
+import frontend.FrontEnd;
+import frontend.handlers;
+import java.awt.Color;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.StringTokenizer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.text.BadLocationException;
+
+/**
+ *
+ * @author Tushar
+ */
+public class step_out implements Runnable
+{
+
+//    public static boolean barrelshiftpresent = false;
+    public static Long pos = new Long(0);
+    public static RandomAccessFile br = null;
+    public static Integer curr_line;
+    public static int common;
+    public Thread t;
+    public ScanFile ob;
+    public File fpath;
+    public static boolean call;
+    public static int initial_pos;
+//    public static int end_pos;
+    public static int next;
+//    private int ld;
+    public static int i = 0;
+    public static int j = 0;
+    private int abc = 0;
+    int exitCondition;
+    //private static boolean commentflag = false;
+
+    public step_out(File path, int ini, int fin, int f)
+      {
+
+        t = new Thread(this, "stepout");
+//        t1 = new Thread(this, "step");
+        fpath = path;
+        exitCondition = ScanFile.call_count;
+        if (f == 0 && ini != 0)
+        {
+
+            StepRun.ini_line = ini;
+            StepRun.fin_line = fin;
+            if (ini == fin)
+            {
+                StepRun.ini_line = 1;
+            } else
+            {
+                StepRun.ini_line = ini;
+            }
+            //System.out.println("BLAH " + StepRun.ini_line + "BLAH" + StepRun.fin_line + "BLAH" + f);
+        }
+
+        f = 1;
+      }
+//    static String[] shifts = {"ROR", "LSL", "LSR", "RRX", "ASR"};
+
+    
+    
+
+    @Override
+    public void run()
+      {
+        try
+        {
+            //System.out.println("step_out run()");
+            scanStart();
+        } catch (InterruptedException ex)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)